### PR TITLE
Add `git merge` statement to merge the targer branch into the source branch

### PR DESCRIPTION
**Pull request by rshad**
Hi all!

This PR resolves https://github.com/wazuh/wazuh-qa/issues/225.

Kr,

Rshad

### CODE REVIEW

File: kitchen/wazuh-puppet/dockerfile [hunk 1]
```diff
@@ -1,14 +1,19 @@
 FROM vpgrp/kitchen:latest
 
 ARG source_branch=master
+ARG target_branch=master
```
Comment (manuasir) [inline kitchen/wazuh-puppet/dockerfile]: Why the same `master` value in both source and target branches?

File: kitchen/wazuh-puppet/dockerfile [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 
 # Update aptitude with new repo
 RUN apt-get update
 
 # Clone the corresponding branch of wazuh-puppet
 RUN git clone https://github.com/wazuh/wazuh-puppet.git
 
-RUN cd wazuh-puppet && git checkout $source_branch && git branch && git pull
+RUN cd wazuh-puppet && \
+    git pull --all && \
```
Comment (manuasir) [inline kitchen/wazuh-puppet/dockerfile]: Why `git pull` here?

File: kitchen/wazuh-puppet/dockerfile (hunk 1 above)
Comment (rshad) [reply kitchen/wazuh-puppet/dockerfile]: Hi Manu!

These are default values, and they will always be overwritten every time we build our image for testing, taking into account the corresponding task in `wazuh-jenkins`:

```yaml
  - name: DOCKER_IMAGE | BUILDING THE REQUIRED DOCKER IMAGE
    docker_image:
      name: "{{ job_name | lower }}-{{  build_num  }}"
      build:
        path: "{{ complete_path }}/kitchen/wazuh-puppet"
        args:
          source_branch: "{{ var_source_branch }}"
          target_branch: "{{ var_target_branch }}"
      source: build
    register: docler_image_output
```

Kr,

Rshad

File: kitchen/wazuh-puppet/dockerfile (hunk 2 above)
Comment (rshad) [reply kitchen/wazuh-puppet/dockerfile]: Hi Manu!

The purpose here is to update the tracking info for all the branches.

Kr,

Rshad

Comment (manuasir) [reply kitchen/wazuh-puppet/dockerfile]: I think we should better use `git fetch --all` here.

**Pull request merged**